lib/fleiss/worker.rb

```diff
@@ -62,34 +60,37 @@ def run_cycle
 
     batch.each do |job|
       @pool.post do
-        Fleiss.backend.wrap_perform { perform(job) }
+        perform(job)
       end
     end
   rescue StandardError => e
-    handle_exception e, 'running cycle'
+    log_exception e, 'running cycle'
   end
 
   def perform(job)
     thread_id = Thread.current.object_id.to_s(16).reverse
     owner     = "#{uuid}/#{thread_id}"
-    return unless job.start(owner)
 
-    log(:info) { "Worker #{uuid} execute job ##{job.id} (by thread #{thread_id})" }
-    finished = false
-    begin
-      ActiveJob::Base.execute job.job_data
-      finished = true
-    rescue StandardError
-      finished = true
-      raise
-    ensure
-      finished ? job.finish(owner) : job.reschedule(owner)
+    Fleiss.backend.wrap_perform do
```

Comment:
Decided to keep it dumb and just wrap what can raise exceptions instead of doing too many changes (extracting perform's method body into a smaller method and rescuing here, then we would have to pass thread/owner to this method - too much hassle, I think).

Reply:
I have no idea what is happening here, except that you moved the wrapper into perform. Nothing seems to have changed

Reply:
Effectively the only change is that now errors are actually handled by Fleiss.backend.wrap_perform, exactly what I was after.